### ISSUE TITLE
error using shiny and slidify together

### ISSUE THREAD

**Issue by WilliamGinWolf** [ramnathv/slidify#491]
Hi, Ramnathv! I am using slidify to create a ppt including Shiny.But now I have encountered with a problem.
Here is my slidify code:
`---
title       : End-Term Presentation
subtitle    : Using Slidify,GoogleVis and Shiny to deal with data visualization
author      : William
job         : Exchange Student
framework   : io2012        # {io2012, html5slides, shower, dzslides, ...}
highlighter : highlight.js,prettify  # {highlight.js, prettify, highlight}
hitheme     : solarized_light      # 
widgets     : [bootstrap,quiz,mathjax,Shiny]            # {mathjax, quiz, bootstrap}
mode        : selfcontained # {standalone, draft}
knit        : slidify::knit2slides
--- &radio
## Questions Before We Start!

Can programming languages doing something cool?
1. _Yes!_
2. No...
3. I don't know

**\* .hint

The answer lies in the Title.

**\* .explanation

This presentation is wriiten in R!

--- ,class #id
## Original Data
- 1.Let's first look at the original data!

``` r opts.label = "code", results = "asis"
setwd("/Users/apple/Documents/i work/about codes/Taiwan/R language/Demo")
original<-read.csv("original.csv")
head(original)
```

--- .class #id
## Dealing With the Data

1.Firstly we use Shiny to write an app which deals with the data foramting problem!

--- .class #id
## Example

``` r,opts.label = 'Shiny'
library(shiny)
library(slidifyLibraries)
library(slidify)
slidifyUI(
  titlePanel("上传文件"),
  sidebarLayout(
    sidebarPanel(
      fileInput('file1', '请上传相应的xls档案',
                ),
      tags$hr(),

      downloadButton('downloadData', 'Download')
    ),
    mainPanel(
      tableOutput('contents')
    )
  )

)
```

--- .class #id
## Shiny And It's APP

``` r, echo = FALSE,cache=FALSE
library(shiny)
library(xlsx)
library(plyr)
shinyApp(
  ui = fluidPage(
    titlePanel("Uploading Data"),
    sidebarLayout(
      sidebarPanel(
        fileInput('file1', 'Please Upload XLSX data',
        ),
        tags$hr(),

        downloadButton('downloadData', 'Download')
      ),
      mainPanel(
        tableOutput('contents')
      )
    )
  ),
  server = function(input, output) {
    output$contents <- renderTable({

      # input$file1 will be NULL initially. After the user selects
      # and uploads a file, it will be a data frame with 'name',
      # 'size', 'type', and 'datapath' columns. The 'datapath'
      # column will contain the local filenames where the data can
      # be found.

      inFile <- input$file1


      if (is.null(inFile))
        return(NULL)


      name<-strsplit(basename(as.character(input$file1)),"[.]")[[1]][1]
      divide(read.xlsx(inFile$datapath,1),name)
    })





    datasetInput <- reactive({
      inFile <- input$file1
      if (is.null(inFile))
        return(NULL)
      name<-strsplit(basename(as.character(input$file1)),"[.]")[[1]][1]
      divide(read.xlsx(inFile$datapath,1),name)
    })

    output$table <- renderTable({
      datasetInput()
    })

    output$downloadData <- downloadHandler(
      filename = function() { 
        name<-strsplit(basename(as.character(input$file1)),"[.]")[[1]][1]
        paste(name, '.csv', sep='') 
      },
      content = function(file) {
        write.csv(datasetInput(), file)
      }
    )

    divide<-function(filename,name){
      totalM = length(which(filename$旅客列车梯形密度表..日均.. == "上车站"))
      daily<-name

      firstTime = 1
      for(i in 1:totalM){
        trainNo <- filename[which(filename$旅客列车梯形密度表..日均.. == "上车站")[i]-1,]
        trainNo <- strsplit(as.character(trainNo[1,1]), " ")[[1]][1]

        a <- filename[which(filename$旅客列车梯形密度表..日均.. == "上车站")[i]:which(filename$旅客列车梯形密度表..日均.. == "上车人数合计")[i],1:which(filename[which(filename$旅客列车梯形密度表..日均.. == "上车站")[i],]=="下车人数合计")]
        colnames(a) <- as.data.frame.array(a[1,]) 
        a <- a[-1,]
        rownames(a) <- a[,1]
        b <- c("ZD111-01","ZD111-02","ZD311","ZD326","ZD192","ZD022","ZD250","ZD062","ZD120","ZD121","ZD143","ZD370","ZD190-02","ZD190-01")

        c2 <- data.frame()
        c2 <- a[,2]
        if(count(names(a) %in% b)[1,2]!=length(names(a) %in% b)){

          if(count(names(a) %in% b)[2,2] >=2 ){
            c2 <- cbind(c2,a[,(names(a) %in% b)])
          }else{
            c2 <- cbind(c2,a[,1:2],a[,(names(a) %in% b)])
            c2 <- c2[,-2]
            c2 <- c2[,-2]
            for(n in 1:length(names(a) %in% b)){
              if((names(a) %in% b)[n]==TRUE){
                break
              }
            }
            colnames(c2)[2] <- colnames(a)[n]
          }

          if(length(c2[1,])>=2){
            for(m in 2:length(c2[1,])){
              c <- c2[,1:2]
              c <- cbind(c,c2[,m])
              colnames(c)[3] <- colnames(c2)[m]
              c <- c[,-2]
              c <- c[!is.na(c[,2]),]

              if(firstTime==1){
                firstTime=0
                d <- t(data.frame(c("当日时间","車次","上車站","下車站","到站時間","離站時間","人數"),
                                  c(name,trainNo,colnames(c)[2],rownames(c)[3],as.character(c[1,2]),as.character(c[3,1]),as.character(c[3,2]))
                ))
                colnames(d) <- d[1,]
              }
              if(length(c[,2])>4){
                d <- rbind(d,c(name,trainNo,colnames(c)[2],rownames(c)[3],as.character(c[1,2]),as.character(c[3,1]),as.character(c[3,2])))
              }

              totalR = length(c[,2]) -4
              if(totalR >0){
                for(j in 1:totalR){
                  d <- rbind(d,c(name,trainNo,colnames(c)[2],rownames(c)[3+j],as.character(c[1,2]),as.character(c[3+j,1]),as.character(c[3+j,2])))
                }
              }

              totalC = length(c[1,]) -2
              if(totalC >0){
                for(k in 1:totalC){
                  c <- c[,-2]
                  c <- c[!is.na(c[,2]),]
                  d <- rbind(d,c(name,trainNo,colnames(c)[2],rownames(c)[3],as.character(c[1,2]),as.character(c[3,1]),as.character(c[3,2])))

                  totalR2 = length(c[,2]) -4
                  if(totalR2 > 0){
                    for(l in 1:totalR2){
                      d <- rbind(d,c(name,trainNo,colnames(c)[2],rownames(c)[3+l],as.character(c[1,2]),as.character(c[3+l,1]),as.character(c[3+l,2])))
                    }
                  }
                }
              }
            }
          }
        }
      }
      d <- d[-1,]
      rownames(d) <- NULL
      return(d)
    }


  }
)

```

--- .class #id

`

But when I use"runback()", the Shiny part cannot run and shows this:

Shiny And It's APP
<iframe data-deferred-src="app245606c15364c72242385f62be0228a0/?w=&subapp_=1" width="100%" height="400" class="shiny-frame shiny-frame-deferred">
How can I deal with this?
Thanks a lot for helping me!
